Added more relaxed check that cloud is polar or cartesian.

michalpelka opened the pull request:
In current implementation the `coordinateNotation` value is checked. It is fine, unless you want to add more fields.

# How this PR was tested

I've run build example an run against multiscan136.

rostest reviewed (commented):
I got the idea.

I would replace it with the following logic:

Cartesian part:

``` 
if (!isCartesian) {
    const char* required[] = {"x", "y", "z", "i"};
    const size_t num_required = sizeof(required) / sizeof(required[0]);
    int found = 0;

    for (size_t i = 0; i < num_required; ++i) {
        for (const auto& field : pointcloud_msg.fields) {
            if (field.name == required[i]) {
                ++found;
                break;
            }
        }
    }

    isCartesian = (found == num_required);
}
```

and for the polar part:

```
if (!isPolar) {
    const char* required[] = {"azimuth", "elevation", "range", "i"};
    const size_t num_required = sizeof(required) / sizeof(required[0]);
    int found = 0;

    for (size_t i = 0; i < num_required; ++i) {
        for (const auto& field : pointcloud_msg.fields) {
            if (field.name == required[i]) {
                ++found;
                break;
            }
        }
    }

    isPolar = (found == num_required);
}
```

michalpelka commented:
Thanks for suggestion @rostest  - I've implemented similar solution:
- moved code to separate method
- utilized std::set.
As previous tested on multiscan136.

rostest reviewed (approved):
nice implementation.
